src/icrar/leap-accelerate/math/linear_math_helper.cc

```diff
@@ -78,4 +91,22 @@ namespace icrar
         }
         return res;
     }
+
+    icrar::MVDirection ToDirection(const casacore::MVDirection& value)
+    {
+        return icrar::MVDirection(value.get()[0], value.get()[1], value.get()[2]);
+    }
+
+    std::vector<icrar::MVDirection> ToDirectionVector(const std::vector<casacore::MVDirection>& value)
+    {
+        // see https://stackoverflow.com/questions/33379145/equivalent-of-python-map-function-using-lambda
+        std::vector<icrar::MVDirection> res(value.size()); //TODO: this populates with 0, O(n), need to reserve and use back_inserter
+        res.reserve(value.size());
+        auto lambda = [](casacore::MVDirection x)
```

Comment:
You already have `ToDirection` above, there doesn't seem to be a need for this lambda

Reply:
that's a good simplification, I might even be able to to make a template that does both the allocation and transform for any vector<T>

Reply:
in a later pull request I add icrar::vector_map(vector, lambda) to help shrink this file further